File: other-languages/sorting/insertionsort/insertionsort.php
<?php
/*
InsertionSort
Autor:
    ?
Colaborador:    
    Maurício Sipmann ( sipmann@gmail.com )
Tipo:
    sorting
Descrição:
    Varre o vetor organizando-o em ordem crescente.
    Só pula para o próximo elemento depois de organizar todos os elementos anteriores.
Complexidade:
    O(n²)
Dificuldade:
    facil
Referências:
    http://pt.wikipedia.org/wiki/Insertion_sort
*/

/*
 * Parâmetros
 * vetor
 *      Vetor de números inteiros
 */
function insertionSort($vetor)
{
    //Vamos começar o loop pelo segundo elemento do vetor
    for($i=1;$i<count($vetor);$i++)
    {

        $anterior = $i-1;

        $proximo = $anterior + 1;

        $troca = true;

        while($troca)
        {
            if($vetor[$anterior] > $vetor[$proximo])
            {
                $tmp = $vetor[$anterior];
                $vetor[$anterior] = $vetor[$proximo];
                $vetor[$proximo] = $tmp;
            }
            else
                $troca = false;

            $anterior--;
            $proximo = $anterior + 1;

            if ($anterior<0)
                $troca = false;
        }
    }
	
	return $vetor;
	
}


$vetor = array(98, 78, 2, 4, 100, -2 );
$vetor = insertionSort($vetor);

for($i=0; $i<count($vetor); $i++)
	echo $vetor[$i].'<br>';

?>

File: other-languages/sorting/insertionsort/insertionsort.rb
# encoding: utf-8

=begin
Insertion Sort
Autor: 
    ?
Tipo:
    sorting
Colaborador:
    Bruno Lara Tavares <bruno.exz@gmail.com>
Descrição:
    Percorre uma lista da esquerda para direita e vai deixando os elementos
    mais a esquerda ordenados à medida que avança pela lista.
Complexidade:
    Pior caso: O(n²)
    Melhor caso: O(n)
Dificuldade:
    Facil
Referencia:
    http://pt.wikipedia.org/wiki/Insertion_sort
=end

class Array
    def insertion_sort!()
        (1..self.size-1).each do |i|
            elemento = self[i]
            j = i - 1

            while j >= 0 && self[j] > elemento
                self[j+1] = self[j]
                j -= 1
                self[j+1] = elemento
            end
        end
        self
    end
end

puts [5,6,8,1].insertion_sort!.inspect
